File: 2015.02/srm649.erl
-module(srm649_div2_lv1).
-export([check/2, test/1]).

check(S, T) when ((length(T) + 1) /= length(S)) -> "Impossible";
check(S, T) ->
  case lists:any ((fun(S2) when S2 == T -> true; (_) -> false end), 
  	  [[S2 || S2 <- delete(S, I)] || I <- lists:seq(0, length(T))]) of
  	true -> "Possible";
  	false -> "Impossible"
  end.

delete(List, N) ->
  {L1, [_|L2]} = lists:split(N, List),
  L1 ++ L2.

test(1) -> check("sunuke", "snuke"); 
test(2) -> check("snuke", "snuke");
test(3) -> check("snuke", "skue");
test(4) -> check("snukent", "snuke");
test(5) -> check("aaaaa", "aaaa");
test(6) -> check("aaaaa", "aaa");
test(7) -> check("topcoder", "tpcoder");
test(8) -> check("singleroundmatch", "singeroundmatc").

File: 2015.02/srm649.hs

check s t 
  | length t + 1 /= length s = "Impossible"
  | False == check_res = "Impossible"
  | otherwise = "Possible"
  where check_res = any (==t) [delete i s | i <- [0..length t]]


delete n xs = let (ys,zs) = splitAt n xs in ys ++ (tail zs)

main = do
  print $ check "sunuke" "snuke"
  print $ check "snuke" "snuke"
  print $ check "snuke" "skue"
  print $ check "snukent" "snuke"
  print $ check "aaaaa" "aaaa"
  print $ check "aaaaa" "aaa"
  print $ check "topcoder" "tpcoder"
  print $ check "singleroundmatch" "singeroundmatc"
